src/pages/organize/[orgId]/journeys/[journeyId]/new.tsx

```diff
@@ -0,0 +1,208 @@
+import { GetServerSideProps } from 'next';
+import Head from 'next/head';
+import { useIntl } from 'react-intl';
+import { Box, Chip, Grid, useTheme } from '@material-ui/core';
+import { useContext, useEffect, useState } from 'react';
+
+import DefaultLayout from 'layout/organize/DefaultLayout';
+import EditTextinPlace from 'components/EditTextInPlace';
+import Header from 'layout/organize/elements/Header';
+import JourneyInstanceSidebar from 'components/organize/journeys/JourneyInstanceSidebar';
+import { organizationResource } from 'api/organizations';
+import { PageWithLayout } from 'types';
+import { scaffold } from 'utils/next';
+import SnackbarContext from 'hooks/SnackbarContext';
+import SubmitCancelButtons from 'components/forms/common/SubmitCancelButtons';
+import { useRouter } from 'next/router';
+import ZetkinAutoTextArea from 'components/ZetkinAutoTextArea';
+import { journeyInstancesResource, journeyResource } from 'api/journeys';
+import { ZetkinJourney, ZetkinPerson } from 'types/zetkin';
+
+const scaffoldOptions = {
+  authLevelRequired: 2,
+  localeScope: [
+    'layout.organize',
+    'misc',
+    'pages.organizeJourneyInstance',
+    'pages.organizeNewJourneyInstance',
+  ],
+};
+
+export const getServerSideProps: GetServerSideProps = scaffold(async (ctx) => {
+  const { orgId, journeyId } = ctx.params!;
+
+  const { state: orgQueryState } = await organizationResource(
+    orgId as string
+  ).prefetch(ctx);
+
+  const { state: journeyQueryState } = await journeyResource(
+    orgId as string,
+    journeyId as string
+  ).prefetch(ctx);
+
+  if (
+    orgQueryState?.status === 'success' &&
+    journeyQueryState?.status === 'success'
+  ) {
+    return {
+      props: {
+        journeyId,
+        orgId,
+      },
+    };
+  } else {
+    return {
+      notFound: true,
+    };
+  }
+}, scaffoldOptions);
+
+interface NewJourneyPageProps {
+  journeyId: string;
+  orgId: string;
+}
+
+const NewJourneyPage: PageWithLayout<NewJourneyPageProps> = ({
+  journeyId,
+  orgId,
+}) => {
+  const [assignees, setAssignees] = useState<ZetkinPerson[]>([]);
+  const [subjects, setSubjects] = useState<ZetkinPerson[]>([]);
+  const [note, setNote] = useState('');
+  const [editedNote, setEditedNote] = useState(false);
+  const [title, setTitle] = useState('');
+
+  const { showSnackbar } = useContext(SnackbarContext);
+
+  const intl = useIntl();
+  const theme = useTheme();
+  const router = useRouter();
+
+  const journeyQuery = journeyResource(orgId, journeyId).useQuery();
+  const journey = journeyQuery.data as ZetkinJourney;
```

Comment:
For one, you can unpack the result of the `useQuery` and rename data in the process like so:

```js
const { data: journey } = journeyResource(orgId, journeyId).useQuery();
```

But, I think that to avoid the typecasting you could use a `ZetkinQuery` and pass in the journeyQuery and access the data using a function as the child. Can explain if that doesn't make sense. But you may not want to use ZetkinQuery, so you can ignore that. Just trying to avoid `as`

And secondly, you can

Reply:
Yeah, that's a good point. This is one of those cases where we _know_ that the data exists, because it will have been retrieved from the prefetched cache. But I guess it doesn't hurt to use `ZetkinQuery` anyway, cause I'm not a fan of `as` either.

Reply:
I switched to using `ZetkinQuery` around the entire thing (because `journey` is used all over the place). What do you think?

Reply:
Yea it's hard to say... but the query does make it safe and guarantees that there will be data there.